tests/cpp_validation_tests/test_validation.cpp

```diff
@@ -37,7 +37,7 @@ class UnsupportedValidationCase : public PowerGridError {
 
 using nlohmann::json;
 
-auto read_file(std::filesystem::path const& path) {
+auto read_file(std::filesystem::path const& path) { // NOLINT(clang-diagnostic-unused-function)
```

Comment:
is this really unused? if so, please delete, but i expect this to be a false positive. can you find the underlying reason?